node/impl/full/eth_utils.go

```diff
@@ -700,7 +702,7 @@ func newEthTxReceipt(ctx context.Context, tx ethtypes.EthTx, lookup *api.MsgLook
 		BlockNumber:      blockNumber,
 		Type:             ethtypes.EthUint64(2),
 		Logs:             []ethtypes.EthLog{}, // empty log array is compulsory when no logs, or libraries like ethers.js break
-		LogsBloom:        ethtypes.EmptyEthBloom[:],
+		LogsBloom:        LogsBloom,
```

Comment:
Should we just get rid of `EmptyEthBloom` entirely, because this is the only place it's referenced?

```suggestion
		LogsBloom:        [EthBloomSize / 8]byte{},
```

Reply:
Keeping it does have the nice effect of describing what it's doing. Maybe the better alternative is to turn these into functions:

`newEmptyEthBloom()` and `newFullEthBloom()` and get rid of the globals. Then we also reduce the need for the awkward `copy`

Reply:
@rvagg Done.